automated maven warfile package fails, manual succeeds

currently using maven 3.8.8. special env vars (previously) necessary:

```
export JAVA_HOME=/usr/lib/jvm/jre-openjdk
export M2_HOME=/opt/maven
export MAVEN_HOME=/opt/maven
export PATH=${M2_HOME}/bin:${PATH}
```

errors are:

```
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.009 s <<< FAILURE! - in edu.harvard.iq.dataverse.settings.ConfigCheckServiceTest$TestDirNotExistent
edu.harvard.iq.dataverse.settings.ConfigCheckServiceTest$TestDirNotExistent.mkdirFails  Time elapsed: 0.007 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <false> but was: <true>
        at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at org.junit.jupiter.api.AssertFalse.failNotFalse(AssertFalse.java:63)
        at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:36)
        at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:31)
        at org.junit.jupiter.api.Assertions.assertFalse(Assertions.java:227)
        at edu.harvard.iq.dataverse.settings.ConfigCheckServiceTest$TestDirNotExistent.mkdirFails(ConfigCheckServiceTest.java:82)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.004 s <<< FAILURE! - in edu.harvard.iq.dataverse.settings.ConfigCheckServiceTest$TestDirNotWritable
[ERROR] edu.harvard.iq.dataverse.settings.ConfigCheckServiceTest$TestDirNotWritable.writeCheckFails  Time elapsed: 0.003 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <false> but was: <true>
        at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at org.junit.jupiter.api.AssertFalse.failNotFalse(AssertFalse.java:63)
        at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:36)
        at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:31)
        at org.junit.jupiter.api.Assertions.assertFalse(Assertions.java:227)
        at edu.harvard.iq.dataverse.settings.ConfigCheckServiceTest$TestDirNotWritable.writeCheckFails(ConfigCheckServiceTest.java:58)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

--- a/defaults/main.yml
+++ b/defaults/main.yml
@@ -244,14 +244,14 @@ dataverse:
   service_email: noreply@dataverse.yourinstitution.edu
   smtp: localhost # or the FQDN of your organization's SMTP relay
   solr:
-    download_url: https://dlcdn.apache.org/solr/solr/9.3.0/solr-9.3.0.tgz
+    download_url: https://archive.apache.org/dist/solr/solr/9.3.0/solr-9.3.0.tgz
     group: solr
     root: /usr/local/solr
     user: solr
     version: 9.3.0
     listen: 127.0.0.1
     checksum: sha256:2a924a776d7c06dea561f12bdb17291dbaecc9e9a06f363d4599acb7fb7bfa71
-  srcdir: /tmp/dataverse
+  srcdir: /home/dataverse/dataverse
   thumbnails: false
   unittests:
     enabled: false

--- a/tasks/dataverse-branch.yml
+++ b/tasks/dataverse-branch.yml
@@ -4,9 +4,21 @@
   debug:
     msg: '##### CLONE BRANCH #####'
 
+- name: ensure service account and group exist
+  ansible.builtin.import_tasks: payara_service_account.yml
+
+- name: ensure target dir exists
+  ansible.builtin.file:
+    path: '{{ dataverse.srcdir }}'
+    state: directory
+    owner: '{{ dataverse.payara.user }}'
+    group: '{{ dataverse.payara.user }}'
+
 - name: clone specified branch
   git:
     repo: "{{ dataverse_repo }}"
     dest: "{{ dataverse.srcdir }}"
     version: "{{ dataverse_branch }}"
   when: dataverse_branch != "release"
+  become: true
+  become_user: '{{ dataverse.payara.user }}'

--- a/tasks/dataverse-build.yml
+++ b/tasks/dataverse-build.yml
@@ -35,7 +35,7 @@
   become: true
   become_user: '{{ dataverse.payara.user }}'
 
-- name: build warfile without tests and custom maven. tail /tmp/dataverse/mvn.out for gory details.
+- name: build warfile without tests and custom maven. tail /home/dataverse/dataverse/mvn.out for gory details.
   shell: "source /etc/profile.d/maven.sh && mvn -T 2C clean package -Dmaven.test.skip=true > mvn.out"
   args:
     chdir: "{{ dataverse.srcdir }}"

--- a/tasks/dataverse-build.yml
+++ b/tasks/dataverse-build.yml
@@ -53,7 +53,7 @@
   become: true
   become_user: '{{ dataverse.payara.user }}'
 
-- name: build warfile with specified tests using custom maven. tail /tmp/dataverse/mvn.out for gory details.
+- name: build warfile with specified tests using custom maven. tail /home/dataverse/dataverse/mvn.out for gory details.
   shell: "source /etc/profile.d/maven.sh && mvn {{ dataverse.unittests.argument }} {{ jacoco_arg }} -T 2C clean package > mvn.out"
   args:
     chdir: "{{ dataverse.srcdir }}"

--- a/tasks/dataverse-build.yml
+++ b/tasks/dataverse-build.yml
@@ -32,6 +32,8 @@
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == false and
         maven.version == 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: build warfile without tests and custom maven. tail /tmp/dataverse/mvn.out for gory details.
   shell: "source /etc/profile.d/maven.sh && mvn -T 2C clean package -Dmaven.test.skip=true > mvn.out"
@@ -39,6 +41,8 @@
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == false and
         maven.version != 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: build warfile with specified tests. tail /tmp/dataverse/mvn.out for gory details.
   shell: "mvn {{ dataverse.unittests.argument }} {{ jacoco_arg }} -T 2C clean package > mvn.out"
@@ -46,6 +50,8 @@
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == true and
         maven.version == 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: build warfile with specified tests using custom maven. tail /tmp/dataverse/mvn.out for gory details.
   shell: "source /etc/profile.d/maven.sh && mvn {{ dataverse.unittests.argument }} {{ jacoco_arg }} -T 2C clean package > mvn.out"
@@ -53,6 +59,8 @@
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == true and
         maven.version != 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: copy warfile over release
   copy:

--- a/tasks/dataverse-install.yml
+++ b/tasks/dataverse-install.yml
@@ -10,6 +10,14 @@
   debug:
     msg: '##### DATAVERSE INSTALLER #####'
 
+# this is necessary for the installer. dataverse-storage.yml does the real work later.
+- name: filesdir must pre-exist
+  ansible.builtin.file:
+    path: '{{ dataverse.filesdirs[0].path }}'
+    state: directory
+    owner: '{{ dataverse.payara.user }}'
+    group: '{{ dataverse.payara.group }}'
+
 - ansible.builtin.import_tasks: dataverse-build.yml
   when: dataverse_branch != "release"
   

--- a/tasks/payara.yml
+++ b/tasks/payara.yml
@@ -133,6 +133,15 @@
     mode: '0750'
     state: directory
 
+- name: workaround for FISH-7665
+  lineinfile:
+    path: '{{ payara_dir }}/glassfish/config/asenv.conf'
+    line: '{{ item }}'
+  with_items:
+    - 'export JAVA_TOOL_OPTIONS'
+    - 'JAVA_TOOL_OPTIONS="-Djdk.util.zip.disableZip64ExtraFieldValidation=true --add-opens=java.base/java.io=ALL-UNNAMED"'
+  when: dataverse.payara.zipurl is match(".*payara-6.*")
+
 - name: so installations can still install payara5 if they want
   template:
     src: payara6.service.j2

--- a/tasks/payara.yml
+++ b/tasks/payara.yml
@@ -35,11 +35,8 @@
   when: payara_zip_download.changed
   notify: enable and restart payara
 
-- name: create payara service group
-  group: name={{ dataverse.payara.group }} state=present
-
-- name: create payara service account
-  user: name={{ dataverse.payara.user }} group={{ dataverse.payara.group }}
+- name: ensure service account and group exist
+  ansible.builtin.import_tasks: payara_service_account.yml
 
 - name: payara service account should own {{ dataverse.payara.domain }}
   file: path={{ payara_dir }}/glassfish/domains/{{ dataverse.payara.domain }} owner={{ dataverse.payara.user }}
@@ -136,16 +133,7 @@
     mode: '0750'
     state: directory
 
-- name: workaround for FISH-7665
-  lineinfile:
-    path: '{{ payara_dir }}/glassfish/config/asenv.conf'
-    line: '{{ item }}'
-  with_items:
-    - 'export JAVA_TOOL_OPTIONS'
-    - 'JAVA_TOOL_OPTIONS="-Djdk.util.zip.disableZip64ExtraFieldValidation=true --add-opens=java.base/java.io=ALL-UNNAMED"'
-  when: dataverse.payara.zipurl is match(".*payara-6.*")
-
-- name: workaround for FISH-7665
+- name: so installations can still install payara5 if they want
   template:
     src: payara6.service.j2
     dest: /usr/lib/systemd/system/payara.service

--- a/tasks/payara_service_account.yml
+++ b/tasks/payara_service_account.yml
@@ -1,0 +1,11 @@
+---
+
+- name: ensure payara service group exists
+  ansible.builtin.group: 
+    name: '{{ dataverse.payara.group }}' 
+    state: present
+
+- name: ensure payara service account exists
+  ansible.builtin.user: 
+    name: '{{ dataverse.payara.user }}' 
+    group: '{{ dataverse.payara.group }}'

--- a/tests/group_vars/jenkins.yml
+++ b/tests/group_vars/jenkins.yml
@@ -242,7 +242,7 @@ dataverse:
     version: 9.3.0
     listen: 127.0.0.1
     checksum: sha256:2a924a776d7c06dea561f12bdb17291dbaecc9e9a06f363d4599acb7fb7bfa71
-  srcdir: /tmp/dataverse
+  srcdir: /home/dataverse/dataverse
   thumbnails: false
   unittests:
     enabled: true

--- a/tests/group_vars/vagrant.yml
+++ b/tests/group_vars/vagrant.yml
@@ -240,14 +240,14 @@ dataverse:
   service_email: noreply@dataverse.yourinstitution.edu
   smtp: localhost
   solr:
-    download_url: https://dlcdn.apache.org/solr/solr/9.3.0/solr-9.3.0.tgz
+    download_url: https://archive.apache.org/dist/solr/solr/9.3.0/solr-9.3.0.tgz
     group: solr
     root: /usr/local/solr
     user: solr
     version: 9.3.0
     listen: 127.0.0.1
     checksum: sha256:2a924a776d7c06dea561f12bdb17291dbaecc9e9a06f363d4599acb7fb7bfa71
-  srcdir: /tmp/dataverse
+  srcdir: /home/dataverse/dataverse
   thumbnails: false
   unittests:
     enabled: true

